hazelcast/src/main/java/com/hazelcast/query/impl/predicates/BetweenPredicate.java

```diff
@@ -31,8 +31,10 @@
  */
 public class BetweenPredicate extends AbstractIndexAwarePredicate {
 
-    Comparable to;
-    Comparable from;
+    protected Comparable from;
+    protected Comparable to;
+    private transient volatile Comparable fromConverted;
```

Comment:
Ok. So effectively it is a caching mechanism; once set it will not change. In that cases volatile makes sense.